packages/shared/src/contexts/BootProvider.tsx

```diff
@@ -269,13 +267,13 @@ export const BootDataProvider = ({
         getRedirectUri={getRedirectUri}
         loadingUser={!dataUpdatedAt || !user}
         loadedUserFromCache={loadedFromCache}
-        visit={bootData?.visit}
+        visit={remoteData?.visit}
```

Comment:
If this actually works, my guess where the fix could be is the usage of remote data over the cached one stored from the state. Let's see.